cmd/fs-v1.go

```diff
@@ -1042,3 +1058,63 @@ func (fs *fsObjects) ListObjectsHeal(bucket, prefix, marker, delimiter string, m
 func (fs *fsObjects) ListBucketsHeal() ([]BucketInfo, error) {
 	return []BucketInfo{}, errors.Trace(NotImplemented{})
 }
+
+// SetBucketPolicies sets policy on bucket
+func (fs *fsObjects) SetBucketPolicies(bucket string, policy policy.BucketAccessPolicy) error {
+	// Parse validate and save bucket policy.
+	policyBytes, err := json.Marshal(policy)
+	if err != nil {
+		return err
+	}
+	// Acquire a write lock on bucket before modifying its configuration.
+	bucketLock := fs.nsMutex.NewNSLock(bucket, "")
+	if err := bucketLock.GetLock(globalOperationTimeout); err != nil {
+		return err
+	}
+	// Release lock
+	defer bucketLock.Unlock()
```

Comment:
Note that we don't need to lock the bucket here. SetBucketPolicies is just like PutObject where we don't hold any lock on the bucket.

Reply:
done.